### PR TITLE
mat: Add CDense type and basic methods

### PR DESCRIPTION
**Pull request by btracey**
Updates #738.

Please take a look.

### CODE REVIEW

File: mat/cdense.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+package mat
```
Comment (kortschak) [inline mat/cdense.go]: Insert blank line here.

File: mat/cdense.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import (
+	"gonum.org/v1/gonum/blas/cblas128"
+)
+
+type CDense struct {
```
Comment (kortschak) [inline mat/cdense.go]: Doc comments.

File: mat/cmatrix.go [hunk 3]
```diff
@@ -32,8 +32,8 @@ type Conjugate struct {
 	CMatrix CMatrix
 }
 
-// At returns the value of the element at row i and column j of the transposed
-// matrix, that is, row j and column i of the Matrix field.
+// At returns the value of the element at row i and column j of the conjugate
+// transposed matrix, that is, row j and column i of the Matrix field.
 func (t Conjugate) At(i, j int) complex128 {
 	z := t.CMatrix.At(j, i)
 	return complex(real(z), -imag(z))
```
Comment (kortschak) [inline mat/cmatrix.go]: At some point we are likely to be importing "math/cmplx". Use `cmplx.Conj`?

File: mat/cdense.go (hunk 1 above)
Comment (btracey) [reply mat/cdense.go]: Done. Sorry.

File: mat/cdense.go (hunk 2 above)
Comment (btracey) [reply mat/cdense.go]: Done.

File: mat/cmatrix.go (hunk 3 above)
Comment (btracey) [reply mat/cmatrix.go]: Fixed the old code.

File: mat/cdense.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat
+
+import (
```
Comment (kortschak) [inline mat/cdense.go]: Single line?

**Pull request merged**